Modern UI Rework

loannlagarde opened the pull request:
Bonjour,

> Arrivé il y a un mois dans la communauté, je suis plutôt discret. Les projets Meshtastic et Alsamesh ont su piquer ma curiosité, et j’aimerais contribuer à ma manière.

Je vous propose ici une **petite amélioration graphique du site**. L’objectif est d’obtenir un **visuel plus moderne et attrayant**.

Le **contenu** n’a pas été modifié. Cependant, j’ai dû **ajouter et adapter certains éléments de la page d’accueil** afin de la rendre plus accrocheuse. Certains points pourront bien sûr être ajustés selon vos retours.

Je suggère de **ne pas fusionner directement cette proposition dans la branche principale**, mais plutôt de créer une branche de travail dédiée pour affiner les modifications ensemble.

À mon goût, il reste encore un peu de travail à faire sur la page d’accueil, notamment concernant les images d’illustration et le composant ``AlsaceShape``. Cependant, je préfère vous soumettre cette proposition à ce stade afin d’éviter un développement inutile.

En espérant que vous apprécierez mon travail.

loannlagarde committed:
Refactor site layout and styling:

- Removed unused `toggleMenu.js`.
- Enhanced dark mode support with updated classes and colors.
- Redesigned sections for better alignment and responsiveness.
- Updated assets, including new icons and SVG elements.
- Simplified header/menu structure with improved links and interaction logic.
- Added animation for visual enhancements in the "AlsaceShape" component.

loannlagarde committed:
Update README.md to use the new rework screenshot.

loannlagarde committed:
Refactor Header component to dynamically determine current route index.

loannlagarde committed:
Refactor StaticPage layout and typography:

- Updated layout structure for improved readability and alignment.
- Enhanced typography styles using `@tailwindcss/typography` plugin.
- Improved visual hierarchy and dark mode support.

loannlagarde committed:
Use backticks for PIN code display in join page documentation.

loannlagarde committed:
Fix Header menu layout on small screens: updated `max-md:col-span-2` to `max-md:col-span-4`.

loannlagarde committed:
Refine StaticPage layout and styling:

- Extend maximum width of content container for better readability.
- Adjust typography breakpoints for improved responsiveness.
- Enhance em tag with better inline-block support.

loannlagarde committed:
Merge pull request #1 from loannlagarde/develop

Develop

Silhm commented on bun.lock:
project uses bun (not npm), please keep it that way

loannlagarde replied:
Il s’agit d'une erreur de ma part, je n'utilise pas bun. Je vous laisserai mettre à jour le lock avec les nouvelles dépendances.

fabienengels replied:
Si ça te déranges pas, installes bun, ça sera plus simple et c'est vite fait : https://bun.com/docs/installation

loannlagarde replied:
Fait dans [9b2a4f8](https://github.com/fabienengels/alsamesh.org/pull/9/commits/9b2a4f80140f6af925354d10618b47a8b801d827)

Silhm commented on package-lock.json:
project uses bun (not npm), please keep it that way

Silhm commented on rework-screenshoot.png:
maybe a simple webp would be enough here, no need for super high-quality png screenshot imho

loannlagarde replied:
La modification a été effectuée, toutefois son impact sur le dépôt me semble limité.

fabienengels replied:
Les images peuvent être optimisées automatiquement lors du build : https://docs.astro.build/en/guides/images/ si jamais

loannlagarde replied:
C'est noté. Dans le cas présent l'image ne sera pas compilée.
> **NB:** c'est la première fois que j'utilise Astro 😉. N’hésitez donc pas à me conseiller.

fabienengels replied:
Je n'ai pas non plus une grande expérience avec Atro mais j'utilise Alsamesh pour explorer de nouvelles solutions :)

Silhm reviewed (commented):
Nice work! 
see my comments in some files

loannlagarde committed:
Restoring the package manager file

loannlagarde committed:
Replace outdated screenshot in README with optimized `readme-image.webp`.

loannlagarde committed:
Update dependencies in `bun.lock`:

- Add `@fortawesome/fontawesome-free` at version `^7.1.0`.
- Upgrade `tailwindcss` to version `^4.1.16`.

loannlagarde committed:
Add dynamic GPS to SVG point mapping in AlsaceShape component:

- Created a `node-points.json` file to store GPS coordinates.
- Implemented `gpsToSvg` function to convert GPS coordinates to SVG space.
- Automated SVG point rendering using mapped coordinates with animation effects.

loannlagarde committed:
Center-align "Vous découvrez Meshtastic ?" header text in Home component.

loannlagarde committed:
Add configuration QR code display in Home component:

- Introduced a QR code dialog with toggle functionality for node configuration.
- Added new styling and layout updates for the configuration section.
- Incorporated a button to display the QR code and a clickable link for alternative access.

loannlagarde commented:
Concernant les photos d’illustration, j’ai quelques idées, mais il serait préférable de faire une sélection d’environ six photos (huit maximum).
Je verrais bien un carrousel comportant, pour chaque image, un titre, une courte description et un crédit.
Il serait préférable d’éviter les captures d’écran, qui sont généralement moins attractives et s’accordent moins bien avec le reste du thème du site.
Nous pourrions également inviter l’ensemble de la communauté, par exemple via un canal Discord dédié pour le dépôt des photos, en utilisant les réactions pour le vote.

loannlagarde commented on src/assets/node-points.json:
N’hésitez pas à me signaler s’il y a des nœuds à ajouter. J’ai réalisé une extraction de ceux que je peux actuellement détecter depuis chez moi.

fabienengels replied:
Pour les noeuds, je suis en train de bosser sur une solution pour les stocker dynamiquement et faire une carte dynamique. La carte sur le site actuelle est plus la pour l'illustration. Te casses pas trop la tête avec ça :)

fabienengels replied:
Le carrousel peut être sympa tant que ça reste responsive design, je n'ai pas encore mis en place d'analytics sur le site mais je pense qu'on a pas mal de visite via mobile.

fabienengels replied:
Pour les photos, on peut organiser un vote mais j'avoue que pour ce carrousel, peut être garder les choses simples et te faire confiance pour la sélection. Ceci dit, tu peux toujours faire un appel pour que les gens envoient leurs meilleures photos sur le Discord.

(Et on peut toujours peut être prévoir une galerie si les photos sont assez nombreuses)

loannlagarde committed:
Add image carousel with shuffle functionality in Home component:

- Implemented a new image carousel using a marquee effect.
- Introduced `shuffle` utility in `ArrayUtils.ts` for randomizing image order.
- Replaced static image grid with dynamically loaded and shuffled images.
- Added new images to the project under `/carrousel` directory.

loannlagarde commented on src/components/Home.astro:
```diff
@@ -1,12 +1,53 @@
+    },
+    {
+        file: image6,
+        title: "@FAB1"
```
j'ai un doute sur celle-là

fabienengels replied:
C'est Nicolas posant à côté de ESPL :)

loannlagarde commented:
Je pense que la branche est assez mature pour être fusionnée. Je vous laisse me faire votre retour.

fabienengels commented on README.md:
```diff
@@ -1,4 +1,4 @@
-# alsamesh.org
```
A ne pas commiter :)

fabienengels reviewed (commented):
Deux petits trucs à corriger mais après, pour moi c'est ok pour le merge :)

loannlagarde committed:
Update readme

loannlagarde committed:
Change the title of image 6 in the carousel to "@ESPL"

loannlagarde commented:
Voila
